### ISSUE TITLE
400 bad request

### ISSUE THREAD

**Issue by Hemsell** [AbsolutePS/AbsoluteAPI#7]
I followed the example and using code conversion to change it to vb.net.
I am able to authenticate successfully, but no matter how I alter the call/query I get a 400. 
Not sure if this is the correct place to post this. If not my apologies.

Dim requestConfig = New RequestConfiguration().WithSelect("Serial,id").WithTop(100)
Dim devices() As ReportingDevice = Await api.ReportingDevices.GetDevices(Of ReportingDevice)(requestConfig)

**Comment by absolute-mattwood** [comment]
@Hemsell sorry this took longer than you probably were hoping for a response.

The issue is most likely the `WithSelect("Serial,id")`. Reason being that the value "Serial" should be "serial". This is because all the properties are lower case.

If this does not fix it please let us know.

**Comment by Hemsell** [comment]
Thank you for your reply. unfortunately that didn't make a difference. I also tried using just withtop(100) without the select,
no matter what I do I get a 400 back. Including changing the api version, region, selecting ("XXXXXXXX") 
If I change the login information I get an error 401. 
Can I hire you for a few hours :)

**Comment by absolute-mattwood** [comment]
@Hemsell is there any way you can share the code with me (even adding me as contributor to private repo) so that I can repro? In a past life I do have a small amount of VB experience, but either way I might be able to find the issue in a few minutes if I have access to the reproduction code.

**Comment by Hemsell** [comment]
Public Shared Async Function AbsoluteDT() As Task(Of ReportingDevice)
        'First, create an AbsoluteAPIConfiguration object:
        Dim configuration = New AbsoluteAPIConfiguration() With {
            .DefaultDomain = "api.absolute.com",
            .ApiVersion = "v2",
            .Region = "usdc",
             .Service = "abs1"
}

        'Then, create an AbsoluteAPIToken object:
        Dim token = New AbsoluteAPIToken() With {
            .Name = "AppName",
            .Token = "Some numbers",
            .Secret = "the secret"
        }
        'Finally, create the AbsoluteAPI object:
        Dim api = New AbsoluteAPI(configuration, token)

        Dim requestConfig = New RequestConfiguration().WithSelect("serial,id").WithTop(100)

        'Dim requestConfig = New RequestConfiguration(10, 0)

        ' Public Sub New(top As Integer, skip As Integer, Optional protocol As String = "https://", Optional orderBy As String = "", Optional filter As String = "", Optional [select] As String = "", Optional applyOData As Boolean = True, Optional parameters As NameValueCollection = Nothing)





        Try

            Dim devices() As ReportingDevice = Await api.ReportingDevices.GetDevices(Of ReportingDevice)(requestConfig)
            For Each Device As ReportingDevice In devices
                MsgBox(Device.Id)
            Next

        Catch ex As Exception
            MsgBox(ex.Message)
        End Try


    End Function

**Comment by Hemsell** [comment]
Thank you!!!

**Comment by Hemsell** [comment]
I looked at the NuGet package manager in Visual studio to see what version I have (2.1.0) and notice on Dependencies it lists NETStandard,Version=v2.0
my project targets .Net core 5

Is that the problem?

**Comment by absolute-mattwood** [comment]
No it is not related to .Net Standard version.

I fixed this by changing the region value from "usdc" to "cadc".

```
Dim configuration = New AbsoluteAPIConfiguration() With {
.DefaultDomain = "api.absolute.com",
.ApiVersion = "v2",
.Region = "cadc",
 .Service = "abs1"
}
```

This now returns results but the type conversion is incorrect.

I changed to the following (I changed message box to Console.WriteLine because I created a quick demo console app to test, you can change it back as needed):

```
Dim devices As IEnumerable(Of ReportingDevice) = Await api.ReportingDevices.GetDevices(Of ReportingDevice)(requestConfig)
For Each Device As ReportingDevice In devices.ToList()
    Console.WriteLine(Device.Id)
Next
```

The key here is using IEnumerable(of ReportingDevice) as that is the default type being returned.


The domain and region configuration (and how they are associated) is discussed lightly in this part of the docs: https://github.com/AbsolutePS/AbsoluteAPI#important-account-region

**_Long story short:_**

- If you login to cc.absolute.com to access your account, you should use "api.absolute.com" domain and "cadc" as the region
- If you login to cc.us.absolute.com to access your account, you should use "api.us.absolute.com" domain and "usdc" as the region
- Any other console access point should contact Absolute support for their respective API domain and region codes (e.g. for EU as it is not officially documented)

Thanks for raising this Hemsell, hope this helps. Have a great weekend!
